modules/core/src/test/scala/ch/epfl/bluebrain/nexus/rdf/RelativeIriSpec.scala

```diff
@@ -79,5 +79,35 @@ class RelativeIriSpec extends WordSpecLike with Matchers with Inspectors with Ei
       val rhs = RelativeIri("a/?q=asd#1").right.value
       Eq.eqv(lhs, rhs) shouldEqual true
     }
+
+    "resolve from base url" in {
+      val base = Url("http://a/b/c/d;p?q").right.value
+      val cases = List(
+        "g"       -> "http://a/b/c/g",
+        "./g"     -> "http://a/b/c/g",
+        "g/"      -> "http://a/b/c/g/",
+        "/g"      -> "http://a/g",
+        "//g"     -> "http://g",
+        "?y"      -> "http://a/b/c/d;p?y",
+        "g?y"     -> "http://a/b/c/g?y",
+        "#s"      -> "http://a/b/c/d;p?q#s",
+        "g#s"     -> "http://a/b/c/g#s",
+        "g?y#s"   -> "http://a/b/c/g?y#s",
+        ";x"      -> "http://a/b/c/;x",
+        "g;x"     -> "http://a/b/c/g;x",
+        "g;x?y#s" -> "http://a/b/c/g;x?y#s",
+        "."       -> "http://a/b/c/",
+        "./"      -> "http://a/b/c/",
+        ".."      -> "http://a/b/",
+        "../"     -> "http://a/b/",
+        "../g"    -> "http://a/b/g",
+        "../.."   -> "http://a/",
+        "../../"  -> "http://a/",
+        "../../g" -> "http://a/g"
```

Comment:
add additional test cases:
*   when base ends with slash
*   when base contains a fragment
*   when the number of `..` segments is higher than the number of segments in the base
*   combine `.` and `..` to see them collapsed at the same time

Reply:
The tests are taken form the RFC. But yes, I can add those others